hikari/api/cache.py

```diff
@@ -816,6 +862,91 @@ def update_emoji(
+
+        !!! note
+            This will skip stickers that are being kept alive by a reference.
+
+        Returns
+        -------
+        CacheView[hikari.snowflakes.Snowflake, hikari.stickers.GuildSticker]
+            A cache view of sticker IDs to objects of the stickers that were
+            removed from the cache.
+        """
+
+    @abc.abstractmethod
+    def clear_stickers_for_guild(
+        self, guild: snowflakes.SnowflakeishOr[guilds.PartialGuild], /
+    ) -> CacheView[snowflakes.Snowflake, stickers.GuildSticker]:
+        """Remove the known custom emoji objects cached for a specific guild.
+
+        Parameters
+        ----------
+        guild : hikari.snowflakes.SnowflakeishOr[hikari.guilds.PartialGuild]
+            Object or ID of the guild to remove the cached sticker objects for.
+
+        !!! note
+            This will skip stickers that are being kept alive by a reference.
+
+        Returns
+        -------
+        CacheView[hikari.snowflakes.Snowflake, hikari.stickers.GuildSticker]
+            A view of sticker IDs to objects of the stickers that were removed
+            from the cache.
+        """
+
+    @abc.abstractmethod
+    def delete_sticker(
+        self, sticker: snowflakes.SnowflakeishOr[stickers.GuildSticker], /
+    ) -> typing.Optional[stickers.GuildSticker]:
+        """Remove a sticker from the cache.
+
+        Parameters
+        ----------
+        sticker : hikari.snowflakes.SnowflakeishOr[hikari.stickers.GuildSticker]
+            Object or ID of the sticker to remove from the cache.
+
+        !!! note
+            This will not delete stickers that are being kept alive by a reference.
+
+        Returns
+        -------
+        typing.Optional[hikari.stickers.GuildSticker]
+            The object of the sticker that was removed from the cache or
+            `builtins.None`.
+        """
+
+    @abc.abstractmethod
+    def set_sticker(self, sticker: stickers.GuildSticker, /) -> None:
+        """Add a sticker to the cache.
+
+        Parameters
+        ----------
+        sticker : hikari.stickers.GuildSticker
+            The object of the sticker to add to the cache.
+        """
+
+    @abc.abstractmethod
+    def update_sticker(
+        self, sticker: guilds.stickers.GuildSticker, /
+    ) -> typing.Tuple[typing.Optional[stickers.GuildSticker], typing.Optional[stickers.GuildSticker]]:
+        """Update a sticker object in the cache.
+
+        Parameters
+        ----------
+        sticker : hikari.stickers.GuildSticker
+            The object of the sticker to update in the cache.
+
+        Returns
+        -------
+        typing.Tuple[typing.Optional[hikari.stickers.GuildSticker], typing.Optional[hikari.stickers.GuildSticker]]
+            A tuple of the old cached sticker object if found (else `builtins.None`)
+            and the new cached sticker object if it could be cached (else
+            `builtins.None`).
+        """
```

Comment:
Since when a sticker is updated, it is done in bulk and we get all the information, do we really need an update sticker method?

Reply:
valid